src/en/manhwalike/src/eu/kanade/tachiyomi/extension/en/manhwalike/Manhwalike.kt

```diff
@@ -0,0 +1,217 @@
+        return if (query.isNotEmpty()) {
+            val requestBody = query.toFormRequestBody()
+            val requestHeaders = headersBuilder().buildApiHeaders(requestBody)
+            POST("$baseUrl/search/html/1", requestHeaders, requestBody)
+        } else {
+            val url = baseUrl.toHttpUrl().newBuilder()
+            filters.forEach { filter ->
+                when (filter) {
+                    is GenreFilter -> filter.toUriPart().let { url.addPathSegment(it) }
+                    else -> {}
+                }
+            }
+            url.addQueryParameter("page", page.toString())
+            return GET(url.build(), headers)
+        }
+    }
+
+    override fun searchMangaSelector() = "ul.normal li"
+
+    override fun searchMangaNextPageSelector() = "ul.pagination li:last-child a"
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val mangas = when {
+            document.select(searchMangaSelector()).isEmpty() -> document.select("ul li").map { element ->
+                searchMangaFromElement(element)
+            }
+            else -> document.select(searchMangaSelector()).map { element ->
+                searchMangaFromElement(element)
+            }
+        }
+        val hasNextPage = searchMangaNextPageSelector().let { selector ->
+            document.select(selector).first()
+        } != null
+        return MangasPage(mangas, hasNextPage)
+    }
+
+    override fun searchMangaFromElement(element: Element): SManga {
+        return SManga.create().apply {
+            title = element.selectFirst("img")!!.attr("alt")
+            thumbnail_url = element.selectFirst("img")?.toOriginal()
+            setUrlWithoutDomain(element.selectFirst("a")!!.attr("href"))
+        }
+    }
+
+    // details
+    override fun mangaDetailsParse(document: Document): SManga {
+        return SManga.create().apply {
+            author = document.selectFirst("div.author a")!!.text()
+            status = document.selectFirst("small:contains(Status) + strong")?.text()?.trim().toStatus()
+            genre = document.select("div.categories a").joinToString { it.text() }
+            description = document.selectFirst("div.summary-block p.about")?.text()
+            thumbnail_url = document.selectFirst("div.fixed-img img")?.absUrl("src")
+        }
+    }
+
+    // chapters
+    override fun chapterListSelector() = "ul.chapter-list li"
+    override fun chapterFromElement(element: Element): SChapter {
+        return SChapter.create().apply {
+            setUrlWithoutDomain(element.selectFirst("a")!!.absUrl("href"))
+            name = element.selectFirst("a")!!.text()
+            date_upload = element.selectFirst(".time")!!.text().toDate()
+        }
+    }
+
+    // pages
+    override fun pageListParse(document: Document): List<Page> {
+        return document.select(".chapter-content .page-chapter img").mapIndexed { i, it ->
```

Comment:
Don't use `it`. The most common are `i/idx/index` and `img/image`. I would say any combination of that is valid, but I personally like `i, img`